cou/apps/auxiliary_subordinate.py

```diff
@@ -33,6 +44,59 @@ def _check_ovn_support(self) -> None:
         """
         OVNSubordinate._validate_ovn_support(self.workload_version)
 
+    def _check_auto_restarts(self) -> None:
+        """No-op, skip check auto restarts option.
+
+        This method override the parent class's `_check_auto_restarts()` method
+        because the parent class's will raise an `ApplicationError` if
+        `enable-auto-restarts` is `True`.
+        """
+
+    def pre_upgrade_steps(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> list[PreUpgradeStep]:
+        """Pre Upgrade steps planning.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate upgrade plan
+        :type units: Optional[list[Unit]]
+        :return: List of pre upgrade steps.
+        :rtype: list[PreUpgradeStep]
+        """
+        # OVN subordinate charm does not have upgrade_step; the "upgrade" is
+        # done during the pre-upgrade step, so we need to put the run deferred
+        # hook step before pre-upgrade step.
+        steps = []
+        if self.config.get("enable-auto-restarts", {}).get("value") is False:
+            steps.extend(
+                self.get_run_deferred_hooks_and_restart_pre_upgrade_step(self.subordinate_units)
+            )
+        steps.extend(super().pre_upgrade_steps(target, units))
+        return steps
+
+    def post_upgrade_steps(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> list[PostUpgradeStep]:
+        """Post Upgrade steps planning.
+
+        Wait until the application reaches the idle state and then check the target workload.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate post upgrade plan
+        :type units: Optional[list[Unit]]
+        :return: List of post upgrade steps.
+        :rtype: list[PostUpgradeStep]
+        """
+        steps = []
+        if self.config.get("enable-auto-restarts", {}).get("value") is False:
+            steps.extend(
+                self.get_run_deferred_hooks_and_restart_post_upgrade_step(self.subordinate_units)
+            )
+        steps.extend(super().post_upgrade_steps(target, units))
+        return steps
```

Comment:
I think this is not enough for subordinates.

Imagine the following scenario:
ovn-chassis is related with nova-compute and we start with the subordinate:

1- You do the upgrade steps for ovn-chasis. 
2 - You upgrade nova-compute (Here is when the packages update and ovn might request to restart)
3 - You 'll leave the cloud without restarting the ovn-chassis

This should be called as a post upgrade step of the principal charm and we need to receive specific ovn-chassis units, otherwise you can create an outage of data-plane.